src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1231,15 +1231,29 @@ internal static bool CanSkipRewriting(
 
             if (!ignoreComReceiver)
             {
-                var receiverNamedType = invokedAsExtensionMethod ?
-                                        ((MethodSymbol)methodOrIndexer).Parameters[0].Type as NamedTypeSymbol :
-                                        methodOrIndexer.ContainingType;
+                NamedTypeSymbol? receiverNamedType = tryGetReceiverNamedType(methodOrIndexer, invokedAsExtensionMethod);
                 isComReceiver = receiverNamedType is { IsComImport: true };
             }
 
             return rewrittenArguments.Length == methodOrIndexer.GetParameterCount() &&
                 argsToParamsOpt.IsDefault &&
                 !isComReceiver;
+
+            static NamedTypeSymbol? tryGetReceiverNamedType(Symbol methodOrIndexer, bool invokedAsExtensionMethod)
+            {
+                if (invokedAsExtensionMethod)
+                {
+                    return ((MethodSymbol)methodOrIndexer).Parameters[0].Type as NamedTypeSymbol;
+                }
+
+                if (methodOrIndexer.GetIsNewExtensionMember())
```

Comment:
>methodOrIndexer.GetIsNewExtensionMember() [](http://example.com/codeflow?start=20&length=41)

Why would this matter for a static case? Is the enclosing method never called for static invocations?

Reply:
Thanks, I hadn't considered that question.
If we have a static call and the receiver type is ComImport, we allow arguments to be passed implicitly by-ref and we do need some argument rewriting.
I'll add the relevant tests where the argument are implicitly passed by-ref or not. Those show that `MakeArguments` creates temps as needed for the implicitly by-ref arguments.

Reply:
> If we have a static call and the receiver type is ComImport, we allow arguments to be passed implicitly by-ref 

Is this an existing behavior for regular static methods?

Reply:
I was not able to find a useful test to illustrate this, but yes all locations that can set the relevant option (`OverloadResolution.Options.AllowRefOmittedArguments`) do it without consideration for instance vs. static.